client/packages/cli/index.js

```diff
@@ -1070,10 +1071,27 @@ async function pushPerms(appId) {
     return;
   }
 
-  const ok = await promptOk(
-    "Pushing permissions rules. This will immediately replace your production rules. OK to proceed?",
-  );
-  if (!ok) return;
+  console.log("Planning...");
+
+  const prodPerms = await fetchJson({
+    path: `/dash/apps/${appId}/perms/pull`,
+    debugName: "Perms pull",
+    errorMessage: "Failed to pull perms.",
+  });
+
+  if (!prodPerms.ok) return;
+
+  const diffedStr = jsonDiff.diffString(prodPerms.data.perms, perms)
+  if (!diffedStr.length) {
+    console.log("There are no changes to apply.");
+    return; 
+  }
+  
+  console.log("The following changes will be applied to your perms:");
+  console.log(diffedStr);
+  
+  const okPush = await promptOk("OK to proceed?");
+  if (!okPush) return;
```

Comment:
There is a race condition in both push schema and perms, where if someone _else_ pushed something in the meantime, we may potentially override it. 

I think we can punt this race condition for now, but perhaps we can do in the future, is to pass in some version signifier, so in case it's not the same, we error out